src/averageAges.js

```diff
@@ -55,7 +69,19 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const children = onlyWithSon
+    ? people.filter(p => p.sex === 'm'
+      && people.some(m => m.name === p.mother))
+    : people.filter(p => people.some(m => m.name
+      === p.mother));
```

Comment:
Using flatMap() and putting everything in the same function:

```suggestion
  const children = people.flatMap((person) => {
    const childMom = people.find((mother) => mother.name === person.mother);
    const child = {
      ...person,
      mom: childMom,
    };

    if (!onlyWithSon) return childMom ? child : [];

    const isSon = person.sex === "m";

    return childMom && isSon ? child : [];
  });
```

Changing the variables names to make it more clear